app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt

```diff
@@ -399,54 +367,85 @@ class MeetingRequestViewModel(
    * the contact is too far away
    */
   fun meetingDistanceCancellation() {
-    val selfProfile = profilesViewModel.selfProfile.value
-    val originPoint = selfProfile?.location ?: GeoPoint(0.0, 0.0)
     updateInboxOfMessages {
       val contactUsers = profilesViewModel.getCancellationMessageProfile()
-      val distances =
-          contactUsers.map {
-            distanceBetweenGeoPoints(it.location ?: GeoPoint(0.0, 0.0), originPoint)
-          }
-      val mapUserDistance = contactUsers.zip(distances).toMap()
-      mapUserDistance.forEach {
-        if (it.value >= FIVE_HUNDRED_METERS_IN_KM) {
-          removeFromMeetingRequestInbox(it.key.uid)
-          removeFromMeetingRequestSent(it.key.uid) {}
-          val targetToken = it.key.fcmToken ?: "null"
-          val targetName = it.key.name
-          setMeetingCancellation(
-              targetToken, CancellationType.DISTANCE, targetName) // targeted to other user
-          sendMeetingCancellation()
-          setMeetingCancellation(senderToken, CancellationType.DISTANCE, it.key.name)
-          sendMeetingCancellation()
-        }
+      val usersInMessagingRange = profilesViewModel.getUsersInMessagingRange()
+      val contactUsersUid = contactUsers.map { it.uid }
+      val usersInMessagingRangeUid = usersInMessagingRange.map { it.uid }
+      val contactUserNotInRangeUid =
+          contactUsersUid.filter { !usersInMessagingRangeUid.contains(it) }
+      val contactUserNotInRange = contactUsers.filter { contactUserNotInRangeUid.contains(it.uid) }
+      contactUserNotInRange.forEach {
+        removeFromMeetingRequestInbox(it.uid)
+        removeFromMeetingRequestSent(it.uid) {}
+        sendCancellationToBothUsers(
+            targetUID = it.uid,
+            targetToken = it.fcmToken ?: "",
+            targetName = it.name,
+            reason = CancellationType.DISTANCE)
       }
     }
   }
 
   /**
-   * Function computing the euclidean distance between two points
+   * Starts a timer to timeout the meeting request in case it takes too long
    *
-   * @param point1: the first point
-   * @param point2: the second point
+   * @param uid: the uid of the user we are sending the meeting request
+   * @param token: the fcm token of the target user
+   * @param context: the current context
    */
```

Comment:
You forgot the name in your documentation.

Reply:
indeed !